core/arch/arm/plat-vexpress/main.c

```diff
@@ -156,7 +156,8 @@ static TEE_Result init_console_itr(void)
 {
 	TEE_Result res = TEE_ERROR_GENERIC;
 
-	res = interrupt_add_main_handler(&console_itr);
+	console_itr.chip = interrupt_get_main_chip();
+	res = interrupt_add_handler(&console_itr);
```

Comment:
Her, i think having a `interrupt_add_main_handler()` wold be nice, to not have this 2 steps: definition/init of `struct itr_handler` outside the function but needed to loading `.chip` field from the function right before calling `interrupt_add_handler()`.

Would you be ok we define an `interrupt_add_main_chip_handler() helper function?

Reply:
I see your point, but I would prefer something more generic. How about
``` C
TEE_Result interrupt_add_handler_with_chip(struct itr_chip *chip,
                                           struct itr_handler *h);
```
Then it could be called as:
``` C
res = interrupt_add_handler_with_chip(interrupt_get_main_chip(),
                                      &console_itr);
```
My point is that the pattern will not differ much if another interrupt controller is used. You just need to supply a different chip pointer.

Reply:
ok, i'll post a fixup commit